code/modules/clothing/glasses/glasses.dm

```diff
@@ -230,3 +232,27 @@
 	icon_state = "redglasses"
 	item_state = "redglasses"
 
+
+/obj/item/clothing/glasses/proc/chameleon(var/mob/user)
+	var/list/glasses_types = typesof(/obj/item/clothing/glasses) - src.type - /obj/item/clothing/glasses/hud - /obj/item/clothing/glasses/night/shadowling - /obj/item/clothing/glasses/hud/toggle - /obj/item/clothing/glasses

```

Comment:
Some of these (not all) can be removed from this line, and solved in the loop below, by checking for 
```if(initial(G.flags) & ABSTRACT```